jerry-core/api/jerry.c

```diff
@@ -3472,28 +3457,31 @@ jerry_get_own_property_descriptor (const jerry_value_t  obj_val, /**< object val
     return false;
   }
 
-  prop_desc_p->is_configurable_defined = true;
-  prop_desc_p->is_configurable = (prop_desc.flags & ECMA_PROP_IS_CONFIGURABLE) != 0;
-  prop_desc_p->is_enumerable_defined = true;
-  prop_desc_p->is_enumerable = (prop_desc.flags & ECMA_PROP_IS_ENUMERABLE) != 0;
-
-  prop_desc_p->is_writable_defined = (prop_desc.flags & ECMA_PROP_IS_WRITABLE_DEFINED) != 0;
-  prop_desc_p->is_writable = prop_desc_p->is_writable_defined ? (prop_desc.flags & ECMA_PROP_IS_WRITABLE) != 0 : false;
-
-  prop_desc_p->is_value_defined = (prop_desc.flags & ECMA_PROP_IS_VALUE_DEFINED) != 0;
-  prop_desc_p->is_get_defined = (prop_desc.flags & ECMA_PROP_IS_GET_DEFINED) != 0;
-  prop_desc_p->is_set_defined = (prop_desc.flags & ECMA_PROP_IS_SET_DEFINED) != 0;
+  prop_desc_p->flags = prop_desc.flags;
+  if ((prop_desc_p->flags & JERRY_PROP_IS_WRITABLE_DEFINED) == 0 || (prop_desc.flags & ECMA_PROP_IS_WRITABLE) == 0)
+  {
+    prop_desc_p->flags &= (uint16_t) ~JERRY_PROP_IS_WRITABLE;
+  }
+  if ((prop_desc_p->flags & JERRY_PROP_IS_CONFIGURABLE_DEFINED) == 0
+      || (prop_desc.flags & ECMA_PROP_IS_CONFIGURABLE) == 0)
```

Comment:
I think the second part of the check is unnecessary. Also true for the other three checks.